csp-mixin/signals/azuresqldb.libsonnet

```diff
@@ -0,0 +1,139 @@
+function(this)
+  {
+    local s = self,
+    filteringSelector: this.filteringSelector,
+    groupLabels: this.groupLabels,
+    instanceLabels: this.instanceLabels,
+    aggLevel: 'none',
+    discoveryMetric: {
+      azuremonitor: 'azure_microsoft_storage_storageaccounts_blobservices_blobcount_average_count',
+    },
+
+    signals: {
+      successfulConns: {
+        name: 'Successful connections by database',
+        description: 'Count of successful connections by database',
+        type: 'gauge',
+        sources: {
+          azuremonitor: {
+            expr: 'label_replace(azure_microsoft_sql_servers_databases_connection_successful_total_count{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)")',
+            legendCustomTemplate: '{{database}}',
+          },
+        },
+      },
+
+      deadlocks: {
+        name: 'Deadlocks by database',
+        description: 'Count of deadlocks by database',
+        type: 'gauge',
+        sources: {
+          azuremonitor: {
+            expr: 'label_replace(azure_microsoft_sql_servers_databases_deadlock_total_count{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)")',
+            legendCustomTemplate: '{{database}}',
+          },
+        },
+      },
+
+      sessions: {
+        name: 'Average sessions by database',
+        description: 'Average number of sessions by database',
+        type: 'gauge',
+        sources: {
+          azuremonitor: {
+            expr: 'label_replace(azure_microsoft_sql_servers_databases_sessions_count_average_count{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)")',
+            legendCustomTemplate: '{{database}}',
+          },
+        },
+      },
+
+      cpuPercent: {
+        name: 'CPU utilization by database',
+        description: 'Percent of CPU utilization by database. If database uses vCores, the vCore percent utilization is shown',
+        type: 'gauge',
+        unit: 'percent',
+        sources: {
+          azuremonitor: {
+            expr: 'label_replace(azure_microsoft_sql_servers_databases_cpu_percent_average_percent{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)")',
+            legendCustomTemplate: '{{database}}',
+          },
+        },
+      },
+
+      vCoreCpuPercent: {
+        name: 'vCore utilization by database',
+        description: 'Percent of CPU utilization by database. If database uses vCores, the vCore percent utilization is shown',
+        type: 'gauge',
+        unit: 'percent',
+        sources: {
+          azuremonitor: {
+            expr: '100 * (label_replace(azure_microsoft_sql_servers_databases_cpu_used_average_count{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)") / label_replace(azure_microsoft_sql_servers_databases_cpu_limit_average_count{%(queriesSelector)s}, "database", "$1", "resourceID", ".+/(.*)"))',
+            legendCustomTemplate: '{{database}} vCore',
+          },
+        },
+      },
```

Comment:
type=gauge  and aggLevel=none is equivalent of type=raw. Right now it would just fine, however, changing aggLevel to instance/group later on would break this signal, as 100 * would be applied in the wrong place.
Best to change type to raw, or move `100 *` to `exprWrappers`:
https://github.com/grafana/jsonnet-libs/tree/master/common-lib/common/signal#configuration-options

Reply:
Ahh excellent. Thank you for this, I'll add it to the exprWrappers.